task/backend/task.go

```diff
@@ -29,3 +33,192 @@ type TaskControlService interface {
 	// AddRunLog adds a log line to the run.
 	AddRunLog(ctx context.Context, taskID, runID influxdb.ID, when time.Time, log string) error
 }
+
+// TaskControlAdaptor creates a TaskControlService for the older TaskStore system.
+// TODO(lh): remove task control adaptor when we transition away from Store.
+func TaskControlAdaptor(s Store, lw LogWriter, lr LogReader) TaskControlService {
+	return &taskControlAdaptor{s, lw, lr}
+}
+
+// taskControlAdaptor adapts a Store and log readers and writers to implement the task control service.
+type taskControlAdaptor struct {
+	s  Store
+	lw LogWriter
+	lr LogReader
+}
+
+func (tcs *taskControlAdaptor) CreateNextRun(ctx context.Context, taskID influxdb.ID, now int64) (RunCreation, error) {
+	return tcs.s.CreateNextRun(ctx, taskID, now)
+}
+
+func (tcs *taskControlAdaptor) FinishRun(ctx context.Context, taskID, runID influxdb.ID) (*influxdb.Run, error) {
+	// Once we completely switch over to the new system we can look at the returned run in the tests.
+	task, err := tcs.s.FindTaskByID(ctx, taskID)
+	if err != nil {
+		return nil, err
+	}
+	tcs.lr.FindRunByID(ctx, task.Org, runID)
+
+	return nil, tcs.s.FinishRun(ctx, taskID, runID)
+}
+
+func (tcs *taskControlAdaptor) CurrentlyRunning(ctx context.Context, taskID influxdb.ID) ([]*influxdb.Run, error) {
+	t, m, err := tcs.s.FindTaskByIDWithMeta(ctx, taskID)
+	if err != nil {
+		return nil, err
+	}
+
+	var rtn = make([]*influxdb.Run, len(m.CurrentlyRunning))
+	for i, cr := range m.CurrentlyRunning {
+		rtn[i] = &influxdb.Run{
+			ID:           influxdb.ID(cr.RunID),
+			TaskID:       t.ID,
+			ScheduledFor: time.Unix(cr.Now, 0).Format(time.RFC3339),
+		}
+		if cr.RequestedAt != 0 {
+			rtn[i].RequestedAt = time.Unix(cr.RequestedAt, 0).Format(time.RFC3339)
+		}
+	}
+	return rtn, nil
+}
+
+func (tcs *taskControlAdaptor) ManualRuns(ctx context.Context, taskID influxdb.ID) ([]*influxdb.Run, error) {
+	t, m, err := tcs.s.FindTaskByIDWithMeta(ctx, taskID)
+	if err != nil {
+		return nil, err
+	}
+
+	var rtn = make([]*influxdb.Run, len(m.CurrentlyRunning))
+	for i, cr := range m.CurrentlyRunning {
+		rtn[i] = &influxdb.Run{
+			ID:           influxdb.ID(cr.RunID),
+			TaskID:       t.ID,
+			ScheduledFor: time.Unix(cr.Now, 0).Format(time.RFC3339),
+		}
+		if cr.RequestedAt != 0 {
+			rtn[i].RequestedAt = time.Unix(cr.RequestedAt, 0).Format(time.RFC3339)
+		}
+	}
+	return rtn, nil
+}
+
+func (tcs *taskControlAdaptor) NextDueRun(ctx context.Context, taskID influxdb.ID) (int64, error) {
+	_, m, err := tcs.s.FindTaskByIDWithMeta(ctx, taskID)
```

Comment:
Even though this code won't be around too long, I guess it could be `FindTaskMetaByID`.